SU2_CFD/include/solvers/CFVMFlowSolverBase.hpp

```diff
@@ -282,6 +1029,18 @@ class CFVMFlowSolverBase : public CSolver {
    */
   void ComputeUnderRelaxationFactor(CSolver** solver, const CConfig* config) final;
 
+  /*!
+   * \brief Set the total residual adding the term that comes from the Dual Time Strategy.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver_container - Container vector with all the solutions.
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] iRKStep - Current step of the Runge-Kutta iteration.
+   * \param[in] iMesh - Index of the mesh in multigrid computations.
+   * \param[in] RunTime_EqSystem - System of equations which is going to be solved.
+   */
+  void SetResidual_DualTime(CGeometry *geometry, CSolver **solver_container, CConfig *config, unsigned short iRKStep,
+                            unsigned short iMesh, unsigned short RunTime_EqSystem) override;
```

Comment:
That is not exactly what happens there, the method is not a template, the class is.
Each flow solver, incomp, nemo, comp, instantiates a different flavor of CFVMFlowSolverBase.
Each template instantiation creates a different SetResidual_DualTime that overrides the CSolver virtual function.

I don't think you can have method templates of virtual functions, as there would be no way to instantiate them, templates and overloading happen at compile time, and virtual functions at runtime.
That is why it is important to use the virtual specifiers (override, final) to guarantee that a method really overrides something (exact same signature, decide which method to call at runtime based on type) rather than overloading it (different signature, including template params, decide what to call at compile time based on arguments).

What you can have is class templates with virtual functions, each class template instantiation is like a totally different class.

Reply:
Yep, thanks for the explanation. https://gcc.godbolt.org/z/T6555e this should better reflect the SU2 reality